modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/inputtype/text/TextLine.ts

```diff
@@ -6,21 +6,28 @@ module api.form.inputtype.text {
     import ValueType = api.data.ValueType;
     import ValueTypes = api.data.ValueTypes;
     import i18n = api.util.i18n;
+    import NumberHelper = api.util.NumberHelper;
+    import DivEl = api.dom.DivEl;
 
     export class TextLine extends support.BaseInputTypeNotManagingAdd<string> {
 
-        private regexpStr: string;
         private regexp: RegExp;
 
+        private maxLength: number;
+
         constructor(config: api.form.inputtype.InputTypeViewContext) {
             super(config);
             this.readConfig(config.inputConfig);
         }
 
         private readConfig(inputConfig: { [element: string]: { [name: string]: string }[]; }): void {
-            let regexpConfig = inputConfig && inputConfig['regexp'] && inputConfig['regexp'][0];
-            let regexp = regexpConfig && regexpConfig['value'];
-            this.regexpStr = regexp || null;
+            const regexpConfig = inputConfig['regexp'] ? inputConfig['regexp'][0] : {};
+            const regexp = regexpConfig ? regexpConfig['value'] : '';
+            this.regexp = new RegExp(regexp);
+
+            const maxLengthConfig = inputConfig['max-length'] ? inputConfig['max-length'][0] : {};
```

Comment:
``max-length`` property from config xml gets converted to ``maxLength`` on the server, so this should be changed to ``inputConfig['maxLength']``